pkg/migration/migrations/retry_jobs_with_exponential_backoff.go

```diff
@@ -0,0 +1,271 @@
+			log.Infof(ctx, "skipping %s operation as the schema change already exists.", opTag)
+			// TODO(sajjad): To discuss: This seems not the best way to ensure that we
+			// are ignoring a mutation. We should have some other direct invariant,
+			// e.g., reading some counter from the table descriptor or in-memory state
+			// that counts the number of times the table has been successfully updated.
+			if d.TestingKnobs != nil && d.TestingKnobs.SkippedMutation != nil {
+				d.TestingKnobs.SkippedMutation()
+			}
+			return nil
+		}
+
+		// Modify the table.
+		log.Infof(ctx, "performing operation: %s", opTag)
+		if _, err := d.InternalExecutor.ExecEx(
+			ctx,
+			fmt.Sprintf("migration-alter-table-%d", tableID),
+			nil, /* txn */
+			sessiondata.InternalExecutorOverride{User: security.NodeUserName()},
+			schemaChangeQuery); err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func readTableDescriptor(
+	ctx context.Context, d migration.TenantDeps, tableID descpb.ID,
+) (catalog.TableDescriptor, error) {
+	var jt catalog.TableDescriptor
+	if err := descs.Txn(ctx, d.Settings, d.LeaseManager, d.InternalExecutor, d.DB, func(
+		ctx context.Context, txn *kv.Txn, descriptors *descs.Collection,
+	) (err error) {
+		jt, err = descriptors.GetImmutableTableByID(ctx, txn, tableID, tree.ObjectLookupFlags{
+			CommonLookupFlags: tree.CommonLookupFlags{
+				AvoidCached: true,
+				Required:    true,
+			},
+		})
+		return err
+	}); err != nil {
+		return nil, err
+	}
+	return jt, nil
+}
+
+// hasColumn return true if storedTable already has the given column. storedTable
+// descriptor must be read from the system as compared to reading from the
+// systemschema package.
+// This function returns an error if the column exists but doesn't match with the
+// table's descriptor defined in systemschema/system.go. The comparison is not strict
+// as several descriptor fields are ignored.
+func hasColumn(storedTable catalog.TableDescriptor, colName string) (bool, error) {
+	storedCol, err := storedTable.FindColumnWithName(tree.Name(colName))
+	if err != nil {
+		if strings.Contains(err.Error(), "does not exist") {
+			return false, nil
+		}
+		return false, err
+	}
+
+	expectedCol, err := systemschema.JobsTable.FindColumnWithName(tree.Name(colName))
+	if err != nil {
+		return false, errors.Wrapf(err, "columns name %s is invalid.", colName)
+	}
+
+	expectedCopy := expectedCol.ColumnDescDeepCopy()
+	storedCopy := storedCol.ColumnDescDeepCopy()
+
+	storedCopy.ID = 0
+	expectedCopy.ID = 0
+
+	if err = ensureProtoMessagesAreEqual(&expectedCopy, &storedCopy); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// hasIndex return true if the table has the given index. It returns
+// an error if the index exists but doesn't match with the table's
+// descriptor defined in systemschema/system.go. The comparison is not strict
+// as several descriptor fields are ignored.
+func hasIndex(storedTable catalog.TableDescriptor, indexName string) (bool, error) {
```

Comment:
can you call this something about the jobs table or pass the jobs table as a parameter?